tests/integrationtests/tst_perfparser.cpp

```diff
@@ -596,6 +596,58 @@ private slots:
         }
     }
 
+    void dump(const Data::BottomUp& bottomUp, QTextStream& stream, const QString& prefix)
+    {
+        stream << prefix << bottomUp.symbol.symbol << QLatin1Char('\n');
+
+        for (const auto& child : bottomUp.children) {
+            dump(child, stream, prefix + QLatin1Char('\t'));
+        }
+    }
+
+    void testCustomCostAggregation_data()
+    {
+        QTest::addColumn<Settings::CostAggregation>("aggregation");
+        QTest::addColumn<QString>("filename");
+
+        QTest::addRow("by symbol") << Settings::CostAggregation::BySymbol << "by_symbol.txt";
+        QTest::addRow("by cpu") << Settings::CostAggregation::ByCPU << "by_cpu.txt";
+        QTest::addRow("by process") << Settings::CostAggregation::ByProcess << "by_process.txt";
+        QTest::addRow("by thread") << Settings::CostAggregation::ByThread << "by_thread.txt";
+    }
+
+    void testCustomCostAggregation()
+    {
+        QFETCH(Settings::CostAggregation, aggregation);
+        QFETCH(QString, filename);
+
+        const auto actualBinaryFile = QFINDTESTDATA("custom_cost_aggregation_testfiles/" + filename);
+
+        Settings::instance()->setCostAggregation(aggregation);
+
+        QStringList perfOptions = {"--call-graph", "dwarf", "--switch-events"};
```

Comment:
please only record the data once on one of the static binaries we ship in perfparser and then analyze the same perf data with different aggregations to see the differences. add the binary perf.data to git then and don't record it in the test. otherwise we will never produce stable tests that can be run on arbitrary machines with arbitrary libc etc.

furthermore, make sure to record with `--sample-cpu`, otherwise the CPU value will always be 0 for all samples.